state-chain/runtime/src/migrations/arbitrum_integration.rs

```diff
@@ -0,0 +1,163 @@
+	};
+	use frame_support::pallet_prelude::*;
+	#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, RuntimeDebug, PartialEq, Eq)]
+	pub struct RuntimeSafeMode {
+		pub emissions: pallet_cf_emissions::PalletSafeMode,
+		pub funding: pallet_cf_funding::PalletSafeMode,
+		pub swapping: pallet_cf_swapping::PalletSafeMode,
+		pub liquidity_provider: pallet_cf_lp::PalletSafeMode,
+		pub validator: pallet_cf_validator::PalletSafeMode,
+		pub pools: pallet_cf_pools::PalletSafeMode,
+		pub reputation: pallet_cf_reputation::PalletSafeMode,
+		pub threshold_signature_ethereum:
+			pallet_cf_threshold_signature::PalletSafeMode<EvmInstance>,
+		pub threshold_signature_bitcoin:
+			pallet_cf_threshold_signature::PalletSafeMode<BitcoinCryptoInstance>,
+		pub threshold_signature_polkadot:
+			pallet_cf_threshold_signature::PalletSafeMode<PolkadotCryptoInstance>,
+		pub broadcast_ethereum: pallet_cf_broadcast::PalletSafeMode<EthereumInstance>,
+		pub broadcast_bitcoin: pallet_cf_broadcast::PalletSafeMode<BitcoinInstance>,
+		pub broadcast_polkadot: pallet_cf_broadcast::PalletSafeMode<PolkadotInstance>,
+		pub witnesser: pallet_cf_witnesser::PalletSafeMode<safe_mode::WitnesserCallPermission>,
+	}
+}
+pub struct ArbitrumIntegration;
+
+impl OnRuntimeUpgrade for ArbitrumIntegration {
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		use cf_chains::{assets::arb::Asset::ArbUsdc, instances::ArbitrumInstance};
+		use frame_support::assert_ok;
+		use frame_system::pallet_prelude::BlockNumberFor;
+		use sp_runtime::traits::Zero;
+		use sp_std::str::FromStr;
+
+		assert_ok!(pallet_cf_environment::RuntimeSafeMode::<Runtime>::translate(
+			|maybe_old: Option<old::RuntimeSafeMode>| {
+				maybe_old.map(|old| {
+					safe_mode::RuntimeSafeMode {
+							emissions: old.emissions,
+							funding: old.funding,
+							swapping: old.swapping,
+							liquidity_provider: old.liquidity_provider,
+							validator: old.validator,
+							pools: old.pools,
+							reputation: old.reputation,
+							threshold_signature_evm: old.threshold_signature_ethereum,
+							threshold_signature_bitcoin: old.threshold_signature_bitcoin,
+							threshold_signature_polkadot: old.threshold_signature_polkadot,
+							broadcast_ethereum: old.broadcast_ethereum,
+							broadcast_bitcoin: old.broadcast_bitcoin,
+							broadcast_polkadot: old.broadcast_polkadot,
+							broadcast_arbitrum: <pallet_cf_broadcast::PalletSafeMode<
+								ArbitrumInstance,
+							> as SafeMode>::CODE_GREEN,
+							witnesser: old.witnesser,
+						}
+				})
+			},
+		));
+
+		let genesis_hash =
+			frame_system::BlockHash::<Runtime>::get(BlockNumberFor::<Runtime>::zero());
+
+		let (
+				key_manager_address,
+				vault_address,
+				address_checker_address,
+				chain_id,
+				usdc_address,
+			): (Address, Address, Address, u64, Address) = if genesis_hash ==
+				// BERGHAIN MAINNET
+				H256::from_str(
+					"0x8b8c140b0af9db70686583e3f6bf2a59052bfe9584b97d20c45068281e976eb9",
```

Comment:
Should we add these somewhere central for use in the future?

Reply:
I've done this in another PR

Reply:
See: #4612